Assets/Scripts/Gameplay/Equipables/LightingStates/Flickering.cs

```diff
@@ -0,0 +1,33 @@
+using UnityEngine;
+using System.Collections;
+
+namespace ProjectWendigo.LightingStates
+{
+    public class Flickering : AState<LightingStateContext>
+    {
+        private float _timer;
+
+        public override void Enter()
+        {
+            GenerateTimer();
+        }
+
+        public override void Update()
+        {
+            if (_timer > 0)
+            {
+                _timer -= Time.deltaTime;
+            }
+            else
+            {
+                this.context.GetComponent<Light>().enabled = !this.context.GetComponent<Light>().enabled;
```

Comment:
Store `Light` component in the state to avoid unnecessary calls to `GetComponent`.